From: YOKOTA Hiroshi <yokota.hgml@gmail.com>
Date: Fri, 24 Nov 2023 14:34:56 +0900
Subject: Use canonical method to invoke sip-build tool

Forwarded: https://github.com/kovidgoyal/calibre/pull/2130

https://www.riverbankcomputing.com/static/Docs/sip/command_line_tools.html
> The build tools can also be invoked using the -m command line
> option of the Python interpreter. For example, the following
> two command lines are equivalent:
>   sip-build -h
>   python -m sipbuild.tools.build -h
---
 setup/build.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/setup/build.py b/setup/build.py
index 42acf21..c00f70a 100644
--- a/setup/build.py
+++ b/setup/build.py
@@ -722,8 +722,8 @@ sip-file = "{os.path.basename(sipf)}"
             self.create_sip_build_skeleton(src_dir, ext)
             cwd = src_dir
             cmd = [
-                sys.executable, '-c',
-                '''from sipbuild.tools.build import main; main();''',
+                sys.executable,
+                '-m', 'sipbuild.tools.build',
                 '--verbose', '--no-make', '--qmake', QMAKE
             ]
         return cmd, sbf, cwd
